### DISCUSSION TITLE
API Key support for pem and produce

### DISCUSSION THREAD

**Discussion by rserentill** [fastlane/fastlane#18106] [Q&A] [answered]
Hello,
Do you guys know when is it planned to provide support for pem and produce with API Key via ConnectAPI?
We have a lot of apps on the store and some scripts that use pem and/or produce. With the new forced 2FA we can't use those anymore, so we need to update our scripts to use API Key authentication instead of username/password. I can see on [the docs](https://docs.fastlane.tools/app-store-connect-api/#supported-actionstools) that is not supported, but I wanted to ask if someone knows when will it be.

Thank you

**Comment by max-ott** [comment, marked as answer]
Please use the search before posting new questions. At least the produce part was already answered a few times :) 

Not all functions are available from Apple: https://developer.apple.com/documentation/appstoreconnectapi is what we can support. The Developer Center produce part is not included in the API. I *guess* the same is valid for pem,
